src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -1775,9 +1775,61 @@ public void connectionIdIsAvailableAfterStart() {
         assertNull(hubConnection.getConnectionId());
     }
 
+    @Test
+    public void connectionTokenAppearsInQSConnectionIdIsOnConnectionInstance() {
+        TestHttpClient client = new TestHttpClient().on("POST", "http://example.com/negotiate?negotiateVersion=1",
+                (req) -> Single.just(new HttpResponse(200, "",
+                                "{\"connectionId\":\"bVOiRPG8-6YiJ6d7ZcTOVQ\"," +
+                                "\"negotiateVersion\": 1," +
+                                "\"connectionToken\":\"connection-token-value\"," +
+                                "\"availableTransports\":[{\"transport\":\"WebSockets\",\"transferFormats\":[\"Text\",\"Binary\"]}]}")));
+
+        MockTransport transport = new MockTransport(true);
+        HubConnection hubConnection = HubConnectionBuilder
+                .create("http://example.com")
+                .withTransportImplementation(transport)
+                .withHttpClient(client)
+                .build();
+
+        assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
+        assertNull(hubConnection.getConnectionId());
+        hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
+        assertEquals(HubConnectionState.CONNECTED, hubConnection.getConnectionState());
+        assertEquals("bVOiRPG8-6YiJ6d7ZcTOVQ", hubConnection.getConnectionId());
+        assertEquals("http://example.com?negotiateVersion=1&id=connection-token-value", transport.getUrl());
+        hubConnection.stop().timeout(1, TimeUnit.SECONDS).blockingAwait();
+        assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
+        assertNull(hubConnection.getConnectionId());
+    }
+
+        @Test
+        public void connectionTokenIsIgnoredIfNegotiateVersionIsNotPresentInNegotiateResponse() {
+            TestHttpClient client = new TestHttpClient().on("POST", "http://example.com/negotiate?negotiateVersion=1",
+                    (req) -> Single.just(new HttpResponse(200, "",
+                            "{\"connectionId\":\"bVOiRPG8-6YiJ6d7ZcTOVQ\"," +
```

Comment:
Should the negotiate response include a connectionToken so we can verify it is ignored?